[Serve] Document how to stream outputs from `@serve.batch` (#36263)
ray-project/ray: doc/source/serve/performance.md

- `max_batch_size` controls the size of the batch.
Once the first request arrives, the batching decorator will wait for a full batch (up to `max_batch_size`) until `batch_wait_timeout_s` is reached. If the timeout is reached, the batch will be sent to the model regardless the batch size.

### Streaming batched requests

> You can stream the outputs from your batched requests with an async generator. Let's convert this `StreamingResponder` class to accept a batch.

Reviewer: ```suggestion
Use an async generator to stream the outputs from your batched requests. Let's convert the `StreamingResponder` class to accept a batch.
```

```{literalinclude} doc_code/batching_guide.py
---
start-after: __single_stream_begin__
end-before: __single_stream_end__
---
```

> You can decorate async generator functions with the {mod}`ray.serve.batch` decorator. Similar to non-streaming methods, the function should take in a `List` of inputs. In each iteration, it should `yield` an iterable of outputs with the same length as the input batch size.

Reviewer: ```suggestion
Decorate async generator functions with the {mod}`ray.serve.batch` decorator. Similar to non-streaming methods, the function takes in a `List` of inputs and in each iteration it `yield`s an iterable of outputs with the same length as the input batch size.
```
